src/historical-stipend-calculator.ts

```diff
@@ -0,0 +1,342 @@
+  });
+  console.log(`Loaded ${records.length} conference records`);
+
+  const results: StipendBreakdown[] = [];
+
+  for (const record of records) {
+    try {
+      const result = await calculateStipend(record);
+      results.push(result);
+    } catch (error) {
+      console.error(`Error processing conference "${record["Event Name"]}":`, error);
+    }
+  }
+
+  // Save all caches to disk
+  distanceCache.saveToDisk();
+  coordinatesCache.saveToDisk();
+  costOfLivingCache.saveToDisk();
+  stipendCache.saveToDisk();
+
+  // Create outputs directory if it doesn't exist
+  const outputsDir = "outputs";
+  if (!fs.existsSync(outputsDir)) {
+    fs.mkdirSync(outputsDir);
+  }
+
+  // Generate filename with timestamp
+  const timestamp = Math.floor(Date.now() / 1000);
+  const outputFile = path.join(outputsDir, `historical_stipends_${timestamp}.csv`);
+
+  // Convert results to CSV
+  const header = [
+    "conference",
+    "location",
+    "conference_start",
+    "conference_end",
+    "flight_departure",
+    "flight_return",
+    "flight_cost",
+    "lodging_cost",
+    "basic_meals_cost",
+    "business_entertainment_cost",
+    "local_transport_cost",
+    "ticket_price",
+    "internet_data_allowance",
+    "incidentals_allowance",
+    "total_stipend",
+  ].join(",");
+
+  const rows = results.map((r) =>
+    [
+      `"${r.conference}"`,
+      `"${r.location}"`,
+      `"${r.conference_start}"`,
+      `"${r.conference_end}"`,
+      `"${r.flight_departure}"`,
+      `"${r.flight_return}"`,
+      r.flight_cost,
+      r.lodging_cost,
+      r.basic_meals_cost,
+      r.business_entertainment_cost,
+      r.local_transport_cost,
+      r.ticket_price,
+      r.internet_data_allowance,
+      r.incidentals_allowance,
+      r.total_stipend,
+    ].join(",")
+  );
+
+  // Write to CSV file
+  fs.writeFileSync(outputFile, [header, ...rows].join("\n"));
+
+  // Clean up temporary CSV
+  fs.unlinkSync(tempCsvPath);
+
+  // Output results to console
+  console.log("Calculation complete. Results saved to:", outputFile);
+  console.table(
+    results.map((r) => ({
+      conference: r.conference,
+      location: r.location,
+      conference_start: r.conference_start,
+      conference_end: r.conference_end,
+      flight_departure: r.flight_departure,
+      flight_return: r.flight_return,
+      flight_cost: r.flight_cost,
+      lodging_cost: r.lodging_cost,
+      basic_meals_cost: r.basic_meals_cost,
+      business_entertainment_cost: r.business_entertainment_cost,
+      local_transport_cost: r.local_transport_cost,
+      ticket_price: r.ticket_price,
+      internet_data_allowance: r.internet_data_allowance,
+      incidentals_allowance: r.incidentals_allowance,
+      total_stipend: r.total_stipend,
+    }))
+  );
+}
+
+// Only run main if this file is being executed directly
+if (import.meta.url === `file://${process.argv[1]}`) {
+  main().catch((err) => {
+    console.error("Execution error:", err);
+  });
+}
```

Comment:
This doesn't work on Windows. I think the right way is:
```suggestion
if (require.main === module) {
  main().catch((err) => {
    console.error("Execution error:", err);
  });
}
```

Reply:
No that's the typescript compiler settings.